### ISSUE TITLE
CrescendoNativeErrorQueue and -ErrorAction Stop

### ISSUE THREAD

**Issue by MTI-gh** [PowerShell/Crescendo#188]
### Summary of the new feature / enhancement

I updated a module to use v1.1 and handle errors using Pop-CrescendoNativeError -EmitAsError. 
The command-line tool quser.exe sends two error messages when a remote machine cannot be reached so this puts two messages in the __CrescendoNativeErrorQueue. 

When using the -ErrorAction Stop parameter on the function it will trigger on the first error as expected. However the second message is still in the ErrorQueue. The next time the command is run, it will spit out that remaining error message even if the command is succesful. 

I've tried to illustrate the behaviour in the screenshots. 
![ErrorFromCLi](https://user-images.githubusercontent.com/59748428/226143912-bffec67a-edf1-45c8-b1ef-cd72640e7627.jpg)
![Example](https://user-images.githubusercontent.com/59748428/226143913-a9a5a9c9-1c7a-447a-892f-953da15c8913.jpg)
drops

### Proposed technical implementation details (optional)

It's open for discussion on whether this is an issue that needs to be solved or if the onus is on us to make changes according to our needs for the module. 
Right now, to solve the problem I tweaked the function Pop-CrescendoNativeError to group multiple messages into 1 and write the errormessage only when the CrescendoNativeErrorQueue is empty.

**Comment by JamesWTruher** [comment]
@MTI-gh 
This also may be associated with `PSNativeCommandUseErrorActionPreference` (could you let me know what that value is for you) - this will sometimes result in *2* error records being posted, which may be causing this behavior. I think for the purposes of crescendo, this variable should probably be set to `false` which means you'll only get 1 error record for each emission of the native app.

I've been thinking that this should be set to `false` by the proxy. If you set it to `false` and run again, does that problem go away?

**Comment by DennisL68** [comment]
> ```json
>            "OutputHandlers": [
>                 {
>                     "Handler": "$input|ConvertFrom-CSV",
>                     "StreamOutput": true,
>                     "ParameterSetName": "Default"
>                 }
>             ]
> ```

quser doesn't support CVS-output.

**Comment by theJasonHelmick** [comment]
thank you @DennisL68 for the information!  Its been a long time since I have used quser -- and if it doesn't;t support structured output (css, json, etc) then you will need to create a custom outputted -- here is an article where Sean explains the process: https://devblogs.microsoft.com/powershell-community/converting-string-output-to-objects/

**Comment by DennisL68** [comment]
Yes, I know (and I already have a working parser for quser.exe) :)

My point was only, that your proposed answer to handle the output of quser.exe, was to set the output mode to "StreamOutput".  
But I assume that is only a feature that works with the built-in structured data handler?

How do we go about to handle this issue as described by @MTI-gh when actually wanting to use `quser.exe`? 

Would a new feature of pre-parsing be a solution in this case? 
I.e. I provide the function to parse the output and then use an option to "pipe it back" as json or csv to the Crescendo output handler?

**Comment by MTI-gh** [comment]
Sorry for the late reply, life's been busy. 

@JamesWTruher I did try with `$PSNativeCommandErrorActionPreference` set to $false but it had the same result. I had to enable this feature with Enable-ExperimentalFeature so I don't think it was playing the issue back when I did the testing in March. When set to $true it did produce an additional error message _Program ended with non-zero exit code: 1_

@theJasonHelmick, @DennisL68  Thanks for the follow-ups!

**Comment by JamesWTruher** [comment]
commit f2a1208d explicitly sets PSNativeCommandErrorActionPreference and I changed the architecture to make the error queue local to the cmdlet rather than being global to the module. That should take care of the `-Stop` issue.

**Comment by theJasonHelmick** [comment]
Thank you! Fixed/closed - in release: https://www.powershellgallery.com/packages/Microsoft.PowerShell.Crescendo/1.1.0-RC1
